### PR TITLE
Specify TaskCreationOptions when using TCS

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/Mvc/issues/8558

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewCompiler.cs [hunk 1]
```diff
@@ -181,7 +181,7 @@ private Task<CompiledViewDescriptor> OnCacheMiss(string normalizedPath)
                     cacheEntryOptions.ExpirationTokens.Add(item.ExpirationTokens[i]);
                 }
 
-                taskSource = new TaskCompletionSource<CompiledViewDescriptor>();
+                taskSource = new TaskCompletionSource<CompiledViewDescriptor>(TaskContinuationOptions.RunContinuationsAsynchronously);
```
Comment (benaadams) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewCompiler.cs]: TaskContinuationOptions.RunContinuationsAsynchronously -> Task**Creation**Options.RunContinuationsAsynchronously

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperService.cs [hunk 2]
```diff
@@ -88,7 +88,7 @@ public async Task<IHtmlContent> ProcessContentAsync(TagHelperOutput output, Cach
                     // There is a small race condition here between TryGetValue and TryAdd that might cause the
                     // content to be computed more than once. We don't care about this race as the probability of
                     // happening is very small and the impact is not critical.
-                    var tcs = new TaskCompletionSource<IHtmlContent>();
+                    var tcs = new TaskCompletionSource<IHtmlContent>(TaskCreationOptions.RunContinuationsAsynchronously);
```
Comment (benaadams) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperService.cs]: TaskContinuationOptions.RunContinuationsAsynchronously -> Task**Creation**Options.RunContinuationsAsynchronously

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs [hunk 3]
```diff
@@ -108,7 +108,7 @@ private async Task<IHtmlContent> CreateCacheEntry(CacheTagKey cacheKey, TagHelpe
             var options = GetMemoryCacheEntryOptions();
             options.AddExpirationToken(new CancellationChangeToken(tokenSource.Token));
             options.SetSize(PlaceholderSize);
-            var tcs = new TaskCompletionSource<IHtmlContent>();
+            var tcs = new TaskCompletionSource<IHtmlContent>(TaskContinuationOptions.RunContinuationsAsynchronously);
```
Comment (benaadams) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs]: TaskContinuationOptions.RunContinuationsAsynchronously -> Task**Creation**Options.RunContinuationsAsynchronously

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewCompiler.cs (hunk 1 above)
Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorViewCompiler.cs]: Yup

**Pull request merged**